substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVTypeRecord.java

```diff
@@ -235,4 +569,1064 @@ public boolean equals(Object obj) {
+            pos = CVUtil.putShort(attrs, buffer, pos);
+            pos = CVUtil.putInt(funcIdx, buffer, pos);
+            /*
+             * TODO: there is some indication the offset is only present if attrs & (MPROP_VIRTUAL |
+             * MPROP_IVIRTUAL) != 0
+             */
+            if ((attrs & (MPROP_VIRTUAL | MPROP_IVIRTUAL)) != 0) {
+                pos = CVUtil.putInt(vtbleOffset, buffer, pos);
+            }
+            pos = CVUtil.putUTF8StringBytes(name, buffer, pos);
+            return pos;
+        }
+
+        @Override
+        public String toString() {
+            return String.format("LF_ONEMETHOD(0x%04x) attr=0x%x(%s) funcIdx=0x%x off=0x%x %s", type, attrs, attrString(attrs), funcIdx, vtbleOffset, name);
+        }
+
+        @Override
+        public int hashCode() {
+            int h = super.hashCode();
+            h = 31 * h + funcIdx;
+            h = 31 * h + vtbleOffset;
+            return h;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (!super.equals(obj)) {
+                return false;
+            }
+            CVOneMethodRecord other = (CVOneMethodRecord) obj;
+            return this.vtbleOffset == other.vtbleOffset && this.funcIdx == other.funcIdx;
+        }
+    }
+
+    static class CVBaseMemberRecord extends FieldRecord {
+
+        private final int basetypeIndex; /* type index of member type */
+        private final int offset; /* in java, usually 0 as there is no multiple inheritance. */
+
+        CVBaseMemberRecord(short attrs, int basetypeIndex, int offset) {
+            super(LF_BCLASS, attrs, "");
+            this.basetypeIndex = basetypeIndex;
+            this.offset = offset;
+        }
+
+        @Override
+        public int computeContents(byte[] buffer, int initialPos) {
+            int pos = CVUtil.putShort(type, buffer, initialPos);
+            pos = CVUtil.putShort(attrs, buffer, pos);
+            pos = CVUtil.putInt(basetypeIndex, buffer, pos);
+            pos = CVUtil.putLfNumeric(offset, buffer, pos);
+            return pos;
+        }
+
+        @Override
+        public String toString() {
+            return String.format("LF_BCLASS(0x%04x) attr=0x%04x(%s ?) baseIdx=0x%04x offset=0x%x", LF_BCLASS, attrs, attrString(attrs), basetypeIndex, offset);
+        }
+
+        @Override
+        public int hashCode() {
+            int h = type;
+            h = 31 * h + attrs;
+            h = 31 * h + basetypeIndex;
+            h = 31 * h + offset;
+            return h;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (!super.equals(obj)) {
+                return false;
+            }
+            CVBaseMemberRecord other = (CVBaseMemberRecord) obj;
+            return this.basetypeIndex == other.basetypeIndex && this.offset == other.offset;
+        }
+    }
+
+    static class CVBaseClassRecord extends CVTypeRecord {
```

Comment:
Actually, it's more like a remnant from the future; it was added (a) for completeness, and (b) because I had a vague idea I might do something clever for interfaces later on. Removed.

Reply:
Gone.